monitor/serie.cc

```diff
@@ -121,51 +121,98 @@ namespace rtm
         }
 
         auto limits = ImPlot::GetPlotLimits();
+
         if (limits.X.Size() < SECTION_SIZE.count())
         {
-            auto it = std::find_if(sections_.begin(), sections_.end(), [&](Section const& s)
-            {
-                return (s.min.count() >  limits.X.Min) and (s.max.count() >= limits.X.Max);
-            });
-            if (it == sections_.end())
-            {
-                // out of scope: show full serie
-                ImPlot::PlotLine(name_.c_str(), &serie_.at(0).x, &serie_.at(0).y, static_cast<int>(serie_.size()),
-                    0, 0, sizeof(Parser::Point));
-                return is_downsampled_;
-            }
+            // Check if we can reuse the cached sections
+            bool can_use_cache = cache_valid_ &&
+                                 (limits.X.Min >= cached_min_) &&
+                                 (limits.X.Max <= cached_max_);
 
-            // display 3 sections (before, here and after if possible)
-            if (it != sections_.begin())
+            if (!can_use_cache)
             {
-                it--;
-            }
+                // Cache miss - need to find the sections
+                cached_sections_.clear();
 
-            for (int i = 0; i < 3; ++i)
-            {
-                Parser::Point const* data = it->points.data();
-                ImPlot::PlotLine(name_.c_str(), &data->x, &data->y, static_cast<int>(it->points.size()),
-                    0, 0, sizeof(Parser::Point));
+                // Find first overlapping section
+                auto it_first = std::find_if(sections_.begin(), sections_.end(), [&](Section const &s)
+                                             { return (s.max.count() >= limits.X.Min) && (s.min.count() <= limits.X.Max); });
+
+                if (it_first == sections_.end())
+                {
+                    // No sections overlap - fallback to downsampled view
+                    cache_valid_ = false;
+                    ImPlot::PlotLine(name_.c_str(), &serie_.at(0).x, &serie_.at(0).y,
```

Comment:
Is displaying data necessary? If we are in this section, it means that the user zoom enough so if the system cannot find data, it probably mean that there is nothing to show in the view no? (and keep the status to normal)